### ISSUE TITLE
use c.Start for Execute

### ISSUE THREAD

**Issue by hallyn** [lxc/go-lxc#83]
The go-lxc bindings are executing 'lxc-execute' for container execute.  Is there a good reason for not using lxccontainer->start() with  useinit=1?

I see a comment in the code referring to an issue with signals, but chasing down the links it seems the problem should have been fixed some time ago in go?

**Comment by brauner** [comment]
@hallyn has a good point. Note that `lxc-execute -n <container-name>` **used** to call `lxc_execute()` internally which is actually an unexported symbol. It should not be accessible by the liblxc command line tools themselves. I've sent a patch to change this so from `LXC 2.0.9` and `LXC 2.1` on `lxc-execute` will call `c->start(c, 1, args)`. This made it way more reliable overall. So `go-lxc` should switch to calling the appropriate start wrapper for execute directly.

tl;dr I agree with @hallyn and `execute()` should use `start()`. :)

**Comment by caglar10ur** [comment]
I believe the signaling issue is still there unless go runtime changed its internals about how it handles cgo. Testing should be easy though, just start using execute and see whether you end up with problems :P

**Comment by brauner** [comment]
@caglar10ur, I read through your old signal handling thread with the GO folks. So if signal fd is the problem then any call to liblxc's start functions would cause trouble since they all use a signal fd. So I guess we can just risk it.

**Comment by brauner** [comment]
Let's just add another API call.
